packages/container/lib/container.js

```diff
@@ -153,18 +159,29 @@ Container.prototype = {
     assert('fullName must be a proper full name', this.registry.validateFullName(normalizedName));
 
     if (options.source) {
-      normalizedName = this.registry.expandLocalLookup(fullName, options);
+      let expandedFullName = this.registry.expandLocalLookup(fullName, options);
       // if expandLocalLookup returns falsey, we do not support local lookup
-      if (!normalizedName) {
-        return;
+      if (!EMBER_MODULE_UNIFICATION) {
+        if (!expandedFullName) {
+          return;
+        }
+
+        normalizedName = expandedFullName;
+      } else if (expandedFullName) {
+        // with ember-module-unification, if expandLocalLookup returns something,
+        // pass it to the resolve without the source
+        normalizedName = expandedFullName;
+        options = copy(options, true);
+        delete options.source;
```

Comment:
This is scenario that will never get hit in practice because it would only be executed if `expandedLocalLookup` is implemented and returns something. Are you suggesting that I do `options = {};`? I'm concerned that some upstream caller might add an addition property to `options` and expect it to be passed to the resolver.